DataFormats/L1TrackTrigger/interface/TTStub.h

```diff
@@ -28,34 +28,80 @@ class TTStub {
   void addClusterRef(edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aTTCluster);
 
   /// Detector element
   DetId getDetId() const { return theDetId; }
   void setDetId(DetId aDetId) { theDetId = aDetId; }
 
-  /// Trigger information
-  double getTriggerDisplacement() const;           /// In FULL-STRIP units! (hence, not implemented herein)
-  void setTriggerDisplacement(int aDisplacement);  /// In HALF-STRIP units!
-  double getTriggerOffset() const;                 /// In FULL-STRIP units! (hence, not implemented herein)
-  void setTriggerOffset(int anOffset);             /// In HALF-STRIP units!
-  double getRealTriggerOffset() const;             /// In FULL-STRIP units! (hence, not implemented herein)
-  void setRealTriggerOffset(float anOffset);       /// In HALF-STRIP units!
+  /// Trigger information; see, e.g., TTStubAlgorithm_official::PatternHitCorrelation for definitions
+  /// Values are passed back from there via the TTStubBuilder, be careful to choose the right one.
+  /// In particular note the difference between values passed as "full" strip units (i.e. the strip
+  /// number or difference between strip numbers) and "half" strip units, which have a 2X finer granularity.
+
+  /// RawBend() [rename of getTriggerDisplacement()]: In FULL strip units!
+  /// Returns the relative displacement between the two cluster centroids, i.e.
+  /// the difference between average row coordinates in inner and outer stack member,
+  /// in terms of outer member pitch (if both pitches are the same, this is just the coordinate difference).
+  /// Flag for rejected stubs: +500 if rejected by FE, +1000 if rejected by CIC chip.
+  double rawBend() const;
+
+  /// setRawBend [rename of setTriggerDisplacement()]: In HALF strip units!
+  /// Sets relative displacement between the two cluster centroids, as above.
+  /// Flag for rejected stubs: +500 if rejected by FE, +1000 if rejected by CIC chip.
+  /// NB: Should probably only be used in TTStubBuilder or very similar.
+  void setRawBend(int aDisplacement);
+
+  /// BendOffset() [rename of getTriggerOffset()]: In FULL strip units!
+  /// Returns the correction offset calculated while accepting/rejecting the stub
+  /// Offset is the projection of a straight line from the beam-spot through the innermost hit
+  /// to the outermost stack member, again in terms of outer member pitch.  It is calculated
+  /// taking the centre of the module at (NROWS/2)-0.5.
+
+  double bendOffset() const;
+
+  /// setBendOffset() [rename of setTriggerOffset()]: In HALF strip units!
+  /// Again restricted to builder code.
+  void setBendOffset(int anOffset);
+
+  /// getRealTriggerOffset(): In FULL strip  units!
+  /// setRealTriggerOffset(): In HALF strip units!
+  /// Removed as obsolete.  Ivan Reid 10/2019
+
+  /// set whether this is a PS module or not;
+  void setModuleType(bool isPSModule);
+
+  /// check if a PS module
+  bool moduleType() const;
```

Comment:
How does one understand that "moduleType() == true" means a PS module ?

Please either use a more descriptive name for the methid (e.g. `isPSModule()`) or return an `enum` with the possible types.

Reply:
fixed.

Reply:
fixed.